pkg/streamingpromql/operators/functions/range_vectors.go

```diff
@@ -634,3 +636,45 @@ func irateIdelta(isRate bool) RangeVectorStepFunction {
 		return resultValue, true, nil, nil
 	}
 }
+
+var QuantileOverTime = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction:                 DropSeriesName,
+	StepFunc:                               quantileOverTime,
+	NeedsSeriesNamesForAnnotations:         true,
+	UseFirstArgumentPositionForAnnotations: true,
+}
+
+func quantileOverTime(step *types.RangeVectorStepData, _ float64, args []types.ScalarData, timeRange types.QueryTimeRange, emitAnnotation types.EmitAnnotationFunc, memoryConsumptionTracker *limiting.MemoryConsumptionTracker) (float64, bool, *histogram.FloatHistogram, error) {
+	if !step.Floats.Any() {
+		return 0, false, nil, nil
+	}
+
+	if step.Floats.Any() && step.Histograms.Any() {
```

Comment:
don't need to check the floats again here

Reply:
Good catch, removed in af919dd